diff --git a/src/propsAllowlists.ts b/src/propsAllowlists.ts
index 39c77a1920dd0312b978e074f498406e8f2694ad..2887f281a92cd2acccaacc6d4bee70997b1c229f 100644
--- a/src/propsAllowlists.ts
+++ b/src/propsAllowlists.ts
@@ -1,4 +1,4 @@
-'use strict';
+"use strict";
 type AllowlistsHolder = {
   UI_THREAD_PROPS_WHITELIST: Record<string, boolean>;
   NATIVE_THREAD_PROPS_WHITELIST: Record<string, boolean>;
@@ -121,5 +121,6 @@ export const PropsAllowlists: AllowlistsHolder = {
     tintColor: true,
     shadowColor: true,
     placeholderTextColor: true,
+    text: true,
   },
 };
